language-plutus-core/bench/Bench.hs

```diff
@@ -57,6 +66,27 @@ main =
                       , bench "writeProgram" $ nf (fmap (serialise . void)) $ parse h
                       ]
 
+                , env evalFiles $ \ ~(f, g) ->
```

Comment:
I'd like to see a note in the code explaining why lazy matching is required. I know it's what the criterion docs suggest to do:

> The function that receives the environment must use lazy pattern matching to deconstruct the tuple (e.g., ~(x, y), not (x, y)), as use of strict pattern matching will cause a crash if an exception-throwing value is passed in.

but

1. do we really care about exceptions here?
2. the docs also say

> Discussion. The environment created in the example above is intentionally not ideal. As Haskell's scoping rules suggest, the variable big is in scope for the benchmarks that use only small. It would be better to create a separate environment for big, so that it will not be kept alive while the unrelated benchmarks are being run.

Reply:
It causes an exception on its own I think. 

To me, this is not that deep - I think it's the only way you can use criterion.